qiskit/transpiler/target.py

```diff
@@ -1048,11 +1051,22 @@ def build_coupling_map(self, two_q_gate=None):
         # existing and return
         if self._coupling_graph is not None:
             cmap = CouplingMap()
-            cmap.graph = self._coupling_graph
+            if filter_idle_qubits:
+                cmap.graph = self._filter_coupling_graph()
+            else:
+                cmap.graph = self._coupling_graph
             return cmap
         else:
             return None
 
+    def _filter_coupling_graph(self):
+        has_operations = set(itertools.chain.from_iterable(self.qargs))
```

Comment:
This is much faster in general. I found three times faster in a quick test with seven edges 
```suggestion
        has_operations = {itertools.chain.from_iterable(self.qargs)}
```

Reply:
Heh, fwiw this isn't really a fast path function (it gets called at most twice during `transpile()`) and in my testing even with `set()` it only takes ~222 microseconds with a 5k element qargs list. But using the `{}` construction is much faster (144 ns vs 222 us) so lets make this change even if nobody will ever notice a performance difference.

Reply:
The reason this is faster is because it's not doing the right thing.  The new form creates a set of a single element (the `chain` object), rather than consuming the flattened iterator of the `qargs`.

Reply:
Good call, I completely skipped over that when I tested the assertion. It is indeed not consuming the iterator reverted in: https://github.com/Qiskit/qiskit-terra/pull/9927/commits/b6808411c094cd919cd59b3551d69419cd807aac

Reply:
No wonder it was so fast! Although it's unfortunate that they have different semantics. (I guess you need a way to say I want a set with one item that is an iterator)  There are linter rules for this. If they are smart enough, it would be nice to just leave it to the linter.